### PR TITLE
Fix - Authentication

### PR DESCRIPTION
**Pull request by Patricknshimiyimana**
# What does this PR do?
- Fix the auth middleware

# Description of Task to be Completed?
- Remove the isLoggedIn check in the protect middleware
- Delete the isLoggedIn user field

## Requirements

- [ ] Require dependences installation
- [ ] Require .env update
- [ ] Documentation updated

# How should this be manually tested?
> - Clone this repository
> - checkout to this branch `fix-auth-middleware`
> - install the dependencies `npm install` and 
> - Run migrations `npm run db:migrate` 
> - Create some users or run seeds `npm run db:seed` and then start the server `npm run dev`
> - Go to `http://localhost:$PORT/api/v1/docs` in your browser to access the api documentation
> - Test the middleware by accessing protected routes.

# Any background context you want to provide?

# What are the relevant pivotal tracker stories?

# Screenshots (if appropriate)

# Questions:
- N/A

### CODE REVIEW

File: src/restful/middlewares/index.js [hunk 1]
```diff
@@ -12,25 +12,22 @@ const protect = async (req, res, next) => {
...
     } catch (error) {
       console.log(error);
-      res.status(401).json({ message: "Not Authorized!" });
+      return res.status(401).json({ error: "Not Authorized!" });
     }
   }
+
   if (!token) {
-    res.status(401).json({ message: "Not Authenticated!" });
+    return res.status(401).json({ error: "Not Authenticated!" });
```
Comment (nicele08) [inline src/restful/middlewares/index.js:30]: @Patricknshimiyimana what do you think if we put this logic above try_catch in order to reduce redundant checks.

Hint:
```javascript
if (!token) {
  return res.status(401).json({ error: "Not Authenticated!" });
}

try {
  // rest content ...

} catch (error) {
  console.log(error);
  return res.status(401).json({ error: "Not Authorized!" });
}

```

**Review by nicele08 — APPROVED** [review]
@Patricknshimiyimana good work, just check my comment before merge.

File: src/restful/middlewares/index.js (hunk 1 above)
Comment (Patricknshimiyimana) [reply src/restful/middlewares/index.js:30]: Hello @nicele08 
That try-catch block is wrapped in an if statement that checks for authorization headers,
And this code runs if there is no token, which would never be the case if place it where you are suggesting.
Check the screenshot below 👇
``` 
if (!token) {
  return res.status(401).json({ error: "Not Authenticated!" });
} 
``` 

image

**Comment by Patricknshimiyimana** [comment]
> @Patricknshimiyimana good work, just check my comment before merge.

File: src/restful/middlewares/index.js (hunk 1 above)
Comment (Patricknshimiyimana) [reply src/restful/middlewares/index.js:30]: > @Patricknshimiyimana what do you think if we put this logic above try_catch in order to reduce redundant checks.
> 
> Hint:
> 
> ```js
> if (!token) {
>   return res.status(401).json({ error: "Not Authenticated!" });
> }
> 
> try {
>   // rest content ...
> 
> } catch (error) {
>   console.log(error);
>   return res.status(401).json({ error: "Not Authorized!" });
> }
> ```

Hello @nicele08 
That try-catch block is wrapped in an if statement that checks for authorization headers,
And this code runs if there is no token, which would never be the case if place it where you are suggesting.
Check the screenshot below 👇
``` 
if (!token) {
  return res.status(401).json({ error: "Not Authenticated!" });
} 
``` 

image

Comment (Patricknshimiyimana) [reply src/restful/middlewares/index.js:30]: @nicele08  That try-catch block is wrapped in an if statement that checks for authorization headers,
And this code runs if there is no token, which would never be the case if place it where you are suggesting.
You gerrit? if you don't Check the screenshot below 👇
``` 
if (!token) {
  return res.status(401).json({ error: "Not Authenticated!" });
} 
``` 
image

**Review by alaintresor — APPROVED** [review]

**Pull request merged**